From 753fb4f24fab059fab6808f9663ed4daba3e2ecc Mon Sep 17 00:00:00 2001
From: Du Chenyang <chenyang.du@intel.com>
Date: Mon, 16 Jul 2012 18:44:51 +0800
Subject: REVERTME Adding build configuration for boot.bin

BZ: 46708

THIS SHOULD BE REPLACED BY https://android-secure.intel.com:8080/#/c/32/
[Note: this link seems to be broken]

Adding an option to support intel boot.bin build, which will disable
the default build of boot.img if TARGET_MAKE_NO_DEFAULT_BOOTIMAGE is true.

[Note: This should be re-worked to remove the 'else' case]

Re-work JIRA: https://otci.intel.com/jira/browse/ADEV-172

Bug: 46708
Category: Device Enablement
Domain: AOSP-build
Origin: Internal
Upstream-candidate: No needs rework

[BOOT CAMP SPECIFIC]

Change-Id: If4a85f757ca8eecce30542b9fd54e7652d1bed3f
Orig-Change-Id: I6b2cf1ad4f63517686085fa95ea6948153e2acbc
Signed-off-by: Sun Yijin <yijin.sun@intel.com>
Signed-off-by: Bruce Beare <bruce.j.beare@intel.com>
Signed-off-by: Du Chenyang <chenyang.du@intel.com>
Reviewed-on: http://android.intel.com:8080/56543
Tested-by: Gao, Shuo <shuo.gao@intel.com>
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
Reviewed-on: http://android.intel.com:8080/74477
Reviewed-by: Yao, Yong <yong.yao@intel.com>
---
 core/Makefile |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index 056e6d0..bf23225 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -426,6 +426,7 @@ ifdef BOARD_KERNEL_PAGESIZE
   INTERNAL_BOOTIMAGE_ARGS += --pagesize $(BOARD_KERNEL_PAGESIZE)
 endif
 
+ifneq ($(TARGET_MAKE_NO_DEFAULT_BOOTIMAGE),true)
 INSTALLED_BOOTIMAGE_TARGET := $(PRODUCT_OUT)/boot.img
 
 ifeq ($(TARGET_BOOTIMAGE_USE_EXT2),true)
@@ -456,6 +457,12 @@ bootimage-nodeps: $(MKBOOTIMG)
 
 endif # TARGET_BOOTIMAGE_USE_EXT2
 
+else # TARGET_MAKE_NO_DEFAULT_BOOTIMAGE
+
+$(INSTALLED_BOOTIMAGE_TARGET): $(MAKE_NO_DEFAULT_BOOTIMAGE_ITEMS)
+	$(MAKE_NO_DEFAULT_BOOTIMAGE)
+endif # TARGET_MAKE_NO_DEFAULT_BOOTIMAGE
+
 else	# TARGET_NO_KERNEL
 # HACK: The top-level targets depend on the bootimage.  Not all targets
 # can produce a bootimage, though, and emulator targets need the ramdisk
-- 
1.7.4.1

